src/core/math_func.hpp

```diff
@@ -308,91 +308,83 @@ int DivideApprox(int a, int b);
 
 /**
  * Computes ceil(a / b) for non-negative a and b.
- * @param a Numerator
- * @param b Denominator
- * @return Quotient, rounded up
+ * @param a Numerator.
+ * @param b Denominator.
+ * @return Quotient, rounded up.
  */
-static inline uint CeilDiv(uint a, uint b)
+static inline uint CeilDiv(const uint a, const uint b)
 {
 	return (a + b - 1) / b;
 }
 
 /**
  * Computes ceil(a / b) * b for non-negative a and b.
- * @param a Numerator
- * @param b Denominator
+ * @param a Numerator.
+ * @param b Denominator.
  * @return a rounded up to the nearest multiple of b.
  */
-static inline uint Ceil(uint a, uint b)
+static inline uint Ceil(const uint a, const uint b)
 {
 	return CeilDiv(a, b) * b;
 }
 
 /**
- * Computes round(a / b) for signed a and unsigned b.
- * @param a Numerator
- * @param b Denominator
- * @return Quotient, rounded to nearest
+ * Computes round(a / b) for integer a and positive integer b.
+ * @param a Numerator.
+ * @param b Denominator.
+ * @return Quotient, rounded to nearest.
  */
-static inline int RoundDivSU(int a, uint b)
+static inline int RoundDivSU(const int a, const uint b)
 {
-	const int _b = static_cast<int>(b);
+	const int b_ = static_cast<int>(b);
```

Comment:
Why not just name this variable right in the commit adding it?

Also, an idea: Check if b >= std::numeric_limits<int>::max and return 0 if it is. That avoids UB casting b to a type that can't fit the value, and will still have the correct result.

Reply:
I wish I'd known about _that_ one earlier 😉.